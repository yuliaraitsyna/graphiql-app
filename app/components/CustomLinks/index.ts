export {GoBackLink} from './GoBack';
export {StartOverLink} from './StartOver';
