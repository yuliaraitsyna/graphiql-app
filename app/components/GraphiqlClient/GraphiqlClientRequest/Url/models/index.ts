import {ChangeEvent} from 'react';

export interface UrlProps {
  name: string;
  placeholder?: string;
  label: string;
  value?: string;
  onFocus?: (e: React.FocusEvent<HTMLInputElement>) => void;
  onBlur?: (e: React.FocusEvent<HTMLInputElement>) => void;
  onInput?: (e: ChangeEvent<HTMLInputElement>) => void;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
}
