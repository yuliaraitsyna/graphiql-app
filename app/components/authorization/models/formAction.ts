export enum FormAction {
  SIGN_IN = 'Sign in',
  SIGN_UP = 'Sign up',
}
