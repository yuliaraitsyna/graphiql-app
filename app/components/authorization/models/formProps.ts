export interface FormProps {
  email: string;
  password: string;
  repeatPassword?: string;
  name?: string;
}
