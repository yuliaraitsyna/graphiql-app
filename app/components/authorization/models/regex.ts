const emailPattern = new RegExp(/^[0-9A-Za-z]{1,}@([-A-Za-z]{1,}\.){1,2}[-A-Za-z]{2,}$/);

export {emailPattern};
