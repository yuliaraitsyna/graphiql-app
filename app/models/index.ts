export interface Team {
  git: string;
  name: string;
  role: string;
  description: string;
  image: string;
}
