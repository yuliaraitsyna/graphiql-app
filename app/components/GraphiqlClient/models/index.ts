import {Header} from '~/components/HeadersEditor/models/header';
import {Variable} from '~/components/models/variable';

export type RequestStatus = 'idle' | 'pending' | 'fullfilled' | 'rejected';

export interface GraphqlRequestState {
  endpointUrl: string;
  sdlUrl: string;
  query: string;
  headers: Header | Header[] | null;
  variables: Variable | Variable[] | null;
}

export const initialStateGraphqlRequest: GraphqlRequestState = {
  endpointUrl: '',
  sdlUrl: '',
  query: '',
  headers: null,
  variables: null,
};
