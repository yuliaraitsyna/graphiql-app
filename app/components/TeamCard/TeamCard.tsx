/* eslint-disable react/prop-types */
import {Avatar, Box, Card, CardContent, Grid, Typography} from '@mui/material';
import {Link} from '@remix-run/react';
import {Team} from '~/models/index';

interface TeamCardProps {
  team: Team;
  key: number;
}

export const TeamCard: React.FC<TeamCardProps> = ({team}) => {
  return (
    <Grid item xs={12} sm={6} md={4}>
      <Box display="flex" height="100%">
        <Card
          style={{flexGrow: 1}}
          sx={{
            flexGrow: 1,
            boxShadow: 3,
            transition: '0.3s',
            '&:hover': {boxShadow: 6},
          }}>
          <CardContent>
            <Link to={team.git}>
              <Avatar src={team.image} sx={{width: 150, height: 150, margin: '0 auto'}} />
            </Link>
            <Typography variant="h6" component="div" align="center">
              {team.name}
            </Typography>
            <Typography variant="body2" color="text.secondary" align="center" sx={{marginBottom: 2}}>
              {team.role}
            </Typography>
            <Typography variant="body1" color="text.secondary" align="center">
              {team.description}
            </Typography>
          </CardContent>
        </Card>
      </Box>
    </Grid>
  );
};
