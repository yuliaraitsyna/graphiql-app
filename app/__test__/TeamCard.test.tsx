import '@testing-library/jest-dom';
import {render} from '@testing-library/react';
import {TeamCard} from '../components/TeamCard';
import {Team} from '../models/index';

jest.mock('@remix-run/react', () => ({
  Link: jest.fn(),
}));

const mockTeam: Team = {
  name: 'Test Team',
  role: 'Developer',
  description: 'This is a test team description.',
  image: 'https://example.com/image.jpg',
  git: 'https://github.com/test-team',
};

describe('Test TeamCard component', () => {
  test('Check if renders team name, role, and description', () => {
    const {getByText} = render(<TeamCard team={mockTeam} key={1} />);
    expect(getByText(mockTeam.name)).toBeInTheDocument();
    expect(getByText(mockTeam.role)).toBeInTheDocument();
    expect(getByText(mockTeam.description)).toBeInTheDocument();
  });

  test('Check if renders correctly', () => {
    const {asFragment} = render(<TeamCard team={mockTeam} key={1} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
