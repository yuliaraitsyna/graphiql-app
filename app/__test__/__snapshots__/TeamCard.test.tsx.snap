// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TeamCard component Check if renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-13i4rnv-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-ab8yd1"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-13p0zo4-MuiPaper-root-MuiCard-root"
        style="flex-grow: 1;"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1kgxqm0-MuiTypography-root"
          >
            Test Team
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-t2mtvq-MuiTypography-root"
          >
            Developer
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-12wjmzy-MuiTypography-root"
          >
            This is a test team description.
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
